### ISSUE TITLE
problem running two instances of sensortag.py in parallel

### ISSUE THREAD

**Issue by voltcode** [IanHarvey/bluepy#25]
I have two copies of sensortag.py - each copy has its own BLE ID it listens to.
If only one Sensor Tag is powered on then its assigned sensortag.py works fine - the other does not because its associated device is not running. This works also if I swap sensotag.pys and devices.

When I launch two devices and one sensortag.py then the program is still behaving as expected.
As soon as I run the second sensortag.py, everything breaks. 

the stack trace I get is:

  File "sensortag.py", line 203, in <module>
    ir, hum, baro, gyro = [ s.read() for s in sensors ]
  File "sensortag.py", line 60, in read
    (rawVobj, rawTamb) = struct.unpack('<hh', self.data.read())
  File "/root/sensors/bluepy/bluepy/btle.py", line 114, in read
    return self.peripheral.readCharacteristic(self.valHandle)
  File "/root/sensors/bluepy/bluepy/btle.py", line 296, in readCharacteristic
    resp = self._getResp('rd')
  File "/root/sensors/bluepy/bluepy/btle.py", line 210, in _getResp
    raise BTLEException(BTLEException.COMM_ERROR, "Error from Bluetooth stack (%s)" % errcode)
btle.BTLEException: Error from Bluetooth stack (comerr)

The only file that I've modified is sensortag.py. The change was tiny and only related to the device ID.

Is bluepy working in some kind of single connection mode? Is there a way I could find the root cause? What would be the best workaround for reading from multiple sensorTags without making big changes to the bluepy internals?

**Comment by voltcode** [comment]
I found a related thread on TI forum, however I do not know how to apply the TI response to bluepy code: http://e2e.ti.com/support/wireless_connectivity/f/538/t/243189 . Maybe someone more knowledgeable in bluepy could make more sense out of it than me ?

**Comment by voltcode** [comment]
I found some hints on the web about some bugs in low 3 kernel versions that prevent bluez from supporting multiple connections in some cases and I decided to try it out.

I switched from Debian Wheezy to Jessie (testing channel) on my VM - ie. kernel has changed from 3.2.0-4-amd64 to 3.16.0-4-amd64. **Everything works fine now** - I have two instances of sensortag.py running in parallel on the same VM sending data and not stopping each other. 

Moreover, Jessie has high bluez version built-in (hcitool says 5.23) therefore there is no need to install bluez 5 manually, but you still have to compile the packages from this repo. 

Debian 8 is coming and perhaps new Raspbian will follow after that. It would be great to modify the source of bluepy later so that manual bluepy 5.4 install is not required.
